src/adapters/ws.rs

```diff
@@ -43,7 +43,7 @@ enum PendingHandshake {
 }
 
 enum RemoteState {
-    WebSocket(WebSocket<ArcTcpStream>),
+    WebSocket(Box<WebSocket<ArcTcpStream>>),
```

Comment:
Why this addition is needed?

Reply:
But this implies a new allocation each time a new connection is accepted. Is there any error that comes from the dependency update that forces this change?

Reply:
But why do you want to decrease memory in the stack by moving it to the heap? Putting things in the heap is slow. It is preferable to not allocate if possible in most cases. 

I do not understand the purpose of this change. Could you explain the benefits you get from this change?

Reply:
ok, you are right. I was thought make the `enum` memory alignment can reduce memory usage, but however let it run faster is more important.

Reply:
I've seen that this comment maybe comes from a `clippy` warning that suggests adding a `Box`. It has sense if the enum is stored with different variants. Nevertheless, the big difference is between the error case and the successful case. It's not expected an error happens, so if wasting more memory in the error case is ok, if we optimize for the successful case, that is where we want to archive the best performance. In this case, the clippy message can be safely ignored.